docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+### 3. Scope
+This specification covers the complete formal definition of the Cadence programming language, including:
+- **Lexical Structure**: The basic building blocks of the language, such as tokens, identifiers, and literals.
+- **Syntax and Grammar**: The rules governing the structure of Cadence programs, including statements, expressions, and declarations.
+- **Type System**: The various data types supported by Cadence, including primitives, resources, structs, and optionals.
+- **Resource Management**: The ownership and lifecycle of resources, ensuring they cannot be copied or deleted incorrectly.
+- **Execution Model**: The runtime behavior of Cadence programs, including how contracts are executed and transactions are processed.
+
+This specification does not cover:
+- **Flow Blockchain Consensus Protocol**: The underlying mechanisms of the Flow blockchain, including its consensus algorithm and network architecture.
+- **Flow-Specific APIs and Services**: Detailed descriptions of APIs or services that interact with the blockchain but are external to the core Cadence language.
+
+## II. Lexical Structure
+
+The lexical structure of Cadence defines the basic building blocks from which Cadence programs are constructed. This section outlines the set of characters allowed in Cadence, how identifiers are named, the use of keywords, literals, and comments.
+
+---
+
+### 1. Character Set
+
+Cadence source code is written using the **Unicode** character set (UTF-8 encoding), which allows for a wide range of characters from various languages. However, the core lexical elements, such as identifiers, operators, and punctuation, are primarily restricted to the ASCII subset of Unicode.
+
+- **Allowed Characters**: 
+  - Letters (a-z, A-Z)
+  - Digits (0-9)
+  - Special symbols used in operators and syntax, such as `+`, `-`, `*`, `/`, `=`, `{`, `}`, `(`, `)`, `[`, `]`, etc.
+  - Whitespace characters: space (`U+0020`), tab (`U+0009`), newline (`U+000A`), and carriage return (`U+000D`).
+
+- **Unicode Support**:
+  - While the core syntax is based on ASCII characters, Unicode is fully supported in **string literals** and **comments**.
+  - Identifiers may include Unicode characters in the ranges of letters and numbers but should generally follow best practices for naming and clarity.
+
+---
+
+### 2. Identifiers
+
+Identifiers in Cadence are used to name **variables**, **constants**, **functions**, **contracts**, **types**, and other user-defined entities. Identifiers must follow specific rules to ensure consistency and avoid conflicts with reserved keywords.
```

Comment:
Contracts are also types, maybe remove. Just like above, we want to be exhaustive/specific, so also remove the remainder of the sentence.

```suggestion
Identifiers in Cadence are used to name **variables**, **constants**, **functions**, and **types**. Identifiers must follow specific rules to ensure consistency and avoid conflicts with reserved keywords.
```

Reply:
Updated.

Reply:
@jsproz Can you please push the update? So far the PR only has one commit. Thanks!